src/modules/todo.js

```diff
@@ -0,0 +1,218 @@
+      index: this.tasks.length + 1,
+      completed: false,
+    });
+    this.saveToLocalStorage();
+    this.render();
+  }
+
+  removeTask(index) {
+    this.tasks = this.tasks.reduce((prev, curr) => {
+      if (curr.index !== index) {
+        curr.index = prev.length + 1;
+        prev.push(curr);
+      }
+      return prev;
+    }, []);
+    this.saveToLocalStorage();
+    this.closeEdit();
+  }
+
+  taskStateChange({ index, completed }) {
+    this.tasks[index - 1].completed = !completed;
+    this.saveToLocalStorage();
+    this.closeEdit();
+  }
+
+  editTask({ index }) {
+    if (this.dragState.dragging) return;
+    this.editIndex = index;
+    this.render();
+  }
+
+  closeEdit() {
+    this.editIndex = -1;
+    this.render();
+  }
+
+  updateTask(value) {
+    if (this.tasks[this.editIndex - 1].description !== value) {
+      this.tasks[this.editIndex - 1].description = value;
+      this.saveToLocalStorage();
+    }
+    this.closeEdit();
+  }
+
+  moveTask({ x, y }) {
+    if (!this.dragState.dragging) return;
+    this.dragState.target.style.top = y - this.dragState.y + 'px';
+    this.dragState.target.style.left = x - this.dragState.x + 'px';
+  }
+
+  reOrderTasks(item, index, steps, direction = 'down') {
+    if (direction === 'up') {
+      let position = index > steps ? index - steps : 1;
+      for (let i = index; i > 1; i -= 1) {
+        if (steps < 1) break;
+        this.tasks[i - 2].index = i;
+        steps -= 1;
+      }
+      this.tasks[index - 1].index = position;
+    } else {
+      let position =
+        index + steps <= this.tasks.length ? index + steps : this.tasks.length;
+      for (let i = index; i < this.tasks.length; i += 1) {
+        if (steps < 1) break;
+        this.tasks[i].index = i;
+        steps -= 1;
+      }
+      this.tasks[index - 1].index = position;
+    }
+    this.tasks = sortObjectArray(this.tasks);
+    this.saveToLocalStorage();
+  }
+
+  handleDrag({ x, y }, task, item, end = false) {
+    if (!end) {
+      if (!this.dragState.dragging) {
+        this.dragState = { target: item, dragging: true, x, y };
+        item.classList.add('dragging');
+        item.onmouseup = (e) => this.handleDrag(e, task, item, true);
+      }
+    } else {
+      if (this.dragState.dragging) {
+        const distance = y - this.dragState.y;
+        const steps = Math.floor(Math.abs(distance) / item.offsetHeight);
+        if (Math.abs(distance) > item.offsetHeight) {
+          if (distance < 0) this.reOrderTasks(item, task.index, steps, 'up');
+          else this.reOrderTasks(item, task.index, steps);
+        }
+        this.dragState = { target: null, dragging: false, x: 0, y: 0 };
+        console.log(item);
+        item.classList.add('no-event');
+        console.log(item.classList);
+        setTimeout(() => {
+          item.classList.remove('no-event');
+        }, 1000);
+        item.classList.remove('dragging');
+      }
+    }
+  }
+
+  saveToLocalStorage() {
```

Comment:
Please use arrow function as it is an ES6 standard.

Reply:
About this one, since I have to have access to `this` keyword inside the function, I am afraid it won't be possible with the arrow function, cause arrow functions don't have a scope